src/FakeItEasy/Core/TypeCatalogue.cs

```diff
@@ -66,14 +74,30 @@ public IEnumerable<Type> GetAvailableTypes()
         [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "Appropriate in try methods.")]
         private static IEnumerable<Assembly> GetAllAssemblies(IEnumerable<string> extraAssemblyFiles)
         {
-            var loadedAssemblies = AppDomain.CurrentDomain.GetAssemblies();
+#if FEATURE_REFLECTION_GETASSEMBLIES
+             var loadedAssemblies = AppDomain.CurrentDomain.GetAssemblies();
+             var loadedAssembliesReferencingFakeItEasy = loadedAssemblies.Where(assembly => assembly.ReferencesFakeItEasy());

```

Comment:
A small point, but I cannot tell why this line needed to be repeated in each block. Could it not have just gone after the `#endif`?